front/src/Phaser/Game/MapEditor/Tools/AreaEditorTool.ts

```diff
@@ -42,20 +44,93 @@ export class AreaEditorTool extends MapEditorTool {
     }
 
     public subscribeToRoomConnection(connection: RoomConnection): void {
-        connection.editMapMessageStream.subscribe((message) => {
-            switch (message.message?.$case) {
+        connection.editMapCommandMessageStream.subscribe((editMapCommandMessage) => {
+            const pendingCommands = this.mapEditorModeManager.getPendingCommands();
+
+            if (pendingCommands.length > 0) {
+                if (pendingCommands[pendingCommands.length - 1].id === editMapCommandMessage.id) {
+                    console.log("OUR NEWEST COMMAND CAME BACK FROM THE BACK, ACKNOWLEDGED!");
+                    pendingCommands.pop();
+                    return;
```

Comment:
I think we should check that the command we get back from the map editor is the oldest command that we sent and was not acknowledged.

So I would rather try something like:

```suggestion
                if (pendingCommands[0].id === editMapCommandMessage.id) {
                    console.log("OUR NEWEST COMMAND CAME BACK FROM THE BACK, ACKNOWLEDGED!");
                    pendingCommands.shift();
                    return;
```